packages/web-components/src/components/va-date/va-date.tsx

```diff
@@ -67,21 +127,40 @@ export class VaDate {
     if (target.classList.contains('input-year')) {
       currentYear = target.value;
     }
-    // Use a leading zero for numbers < 10
-    const numFormatter = new Intl.NumberFormat('en-US', {
-      minimumIntegerDigits: 2,
-    });
 
-    // If Month or Date Select are chosen as empty return ''
-    // Otherwise convert numbers less than 10 to have a leading 0
-    this.value = `${currentYear}-${
-      currentMonth ? numFormatter.format(parseInt(currentMonth)) : ''
-    }-${currentDay ? numFormatter.format(parseInt(currentDay)) : ''}`;
+    this.value = `${currentYear}-${currentMonth ? currentMonth : ''}-${
+      currentDay ? currentDay : ''
+    }`;
 
     // This event should always fire to allow for validation handling
     this.dateChange.emit(event);
   };
 
+  private handleDateKey = (event: KeyboardEvent) => {
+    // Allow 0-9, Backspace, Delete, Left and Right Arrow, and Tab to clear data or move to next field
+    const validKeys = [
+      '0',
+      '1',
+      '2',
+      '3',
+      '4',
+      '5',
+      '6',
+      '7',
+      '8',
+      '9',
+      'Backspace',
+      'ArrowRight',
+      'ArrowLeft',
+      'Tab',
+      'Delete',
+    ];
+    if (validKeys.indexOf(event.key) < 0) {
+      event.preventDefault();
+      return false;
+    }
+  };
```

Comment:
Are these functions separate for a reason? If not, could it live in the `date-utils` file and be shared by both components?

https://github.com/department-of-veterans-affairs/component-library/blob/4d6fe7b8883f254b08b0770dc14eca35efb461a5/packages/web-components/src/components/va-date-text-input/va-date-text-input.tsx#L139-L162

Reply:
Yeah good call I'll move it into date-utils